.config/webpack/webpack.config.ts

```diff
@@ -2,200 +2,271 @@
+import { type Configuration, BannerPlugin } from 'webpack';
+import LiveReloadPlugin from 'webpack-livereload-plugin';
+import VirtualModulesPlugin from 'webpack-virtual-modules';
 
-import { getPackageJson, getPluginJson, hasReadme, getEntries } from './utils';
-import { SOURCE_DIR, DIST_DIR } from './constants';
+import { BuildModeWebpackPlugin } from './BuildModeWebpackPlugin';
+import { DIST_DIR, SOURCE_DIR } from './constants';
+import { getCPConfigVersion, getEntries, getPackageJson, getPluginJson, hasReadme, isWSL } from './utils';
 
 const pluginJson = getPluginJson();
+const cpVersion = getCPConfigVersion();
 
-const config = async (env): Promise<Configuration> => ({
-  cache: {
-    type: 'filesystem',
-    buildDependencies: {
-      config: [__filename],
+const virtualPublicPath = new VirtualModulesPlugin({
+  'node_modules/grafana-public-path.js': `
+import amdMetaModule from 'amd-module';
+
+__webpack_public_path__ =
+  amdMetaModule && amdMetaModule.uri
+    ? amdMetaModule.uri.slice(0, amdMetaModule.uri.lastIndexOf('/') + 1)
+    : 'public/plugins/${pluginJson.id}/';
+`,
+});
+
+const config = async (env): Promise<Configuration> => {
+  const baseConfig: Configuration = {
+    cache: {
+      type: 'filesystem',
+      buildDependencies: {
+        config: [__filename],
+      },
     },
-  },
-
-  context: path.join(process.cwd(), SOURCE_DIR),
-
-  devtool: env.production ? 'source-map' : 'eval-source-map',
-
-  entry: await getEntries(),
-
-  externals: [
-    'lodash',
-    'jquery',
-    'moment',
-    'slate',
-    'emotion',
-    '@emotion/react',
-    '@emotion/css',
-    'prismjs',
-    'slate-plain-serializer',
-    '@grafana/slate-react',
-    'react',
-    'react-dom',
-    'react-redux',
-    'redux',
-    'rxjs',
-    'react-router',
-    'react-router-dom',
-    'd3',
-    'angular',
-    '@grafana/ui',
-    '@grafana/runtime',
-    '@grafana/data',
-
-    // Mark legacy SDK imports as external if their name starts with the "grafana/" prefix
-    ({ request }, callback) => {
-      const prefix = 'grafana/';
-      const hasPrefix = (request) => request.indexOf(prefix) === 0;
-      const stripPrefix = (request) => request.substr(prefix.length);
-
-      if (hasPrefix(request)) {
-        return callback(undefined, stripPrefix(request));
-      }
-
-      callback();
+
+    context: path.join(process.cwd(), SOURCE_DIR),
+
+    devtool: env.production ? 'source-map' : 'eval-source-map',
+
+    entry: await getEntries(),
+
+    externals: [
+      // Required for dynamic publicPath resolution
+      { 'amd-module': 'module' },
+      'lodash',
+      'jquery',
+      'moment',
+      'slate',
+      'emotion',
+      '@emotion/react',
+      '@emotion/css',
+      'prismjs',
+      'slate-plain-serializer',
+      '@grafana/slate-react',
+      'react',
+      'react-dom',
+      'react-redux',
+      'redux',
+      'rxjs',
+      'react-router',
```

Comment:
It might seems obvious but it took me a while to understand why importing from "react-router-dom" in the code base was yielding `undefined`: `react-router-dom` was still marked as external here. I still don't understand why it was the case after executing `yarn create @grafana/plugin update`...